File: API/Models/Vehicle.cs
namespace API.Models;

public class Vehicle
{
    public int Id { get; set; }
    public string Model { get; set; } = string.Empty;
    public string Color { get; set; } = string.Empty;
    public int Year { get; set; }

    public int BrandId { get; set; }
    public Brand Brand { get; set; } = new Brand();

    public virtual ICollection<Photo> Photos { get; set; } = [];
}

File: client/src/app/models/vehicle.ts
import { Photo } from "src/app/models/photo";
import { SelectOption } from "src/app/models/selectOption";

export class Vehicle {
  id: number | null = null;
  model: string | null = null;
  year: number | null = null;
  color: string | null = null;
  brand: SelectOption | null = null;
  photos: Photo[] = [];


  constructor(init?: Partial<Vehicle>) {
    Object.assign(this, init);
  }
}
